experimental/rest_client.go

```diff
@@ -0,0 +1,90 @@
+package experimental
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+)
+
+// Client interface
+type Client interface {
+	Fetch(ctx context.Context, uriPath, uriQuery string) ([]byte, error)
+	Get(ctx context.Context, uriPath, uriQuery string) (*http.Response, error)
+}
+
+type restClient struct {
+	url     string
+	headers map[string]string
+	client  *http.Client
+}
+
+// NewRestClient ...
+func NewRestClient(url string, headers map[string]string) Client {
```

Comment:
@ryantxu What do you think about my suggestion of renaming this to `NewClient`, since the REST part is implied (and the return type is `Client`)?

```suggestion
// NewRestClient creates a Client.
func NewRestClient(url string, headers map[string]string) Client {
```

Reply:
@aknuds1 the reason I didn't do that was if we had some other type of client in the same package.  We could only have one NewClient function (unless we split them into separate packages).

Reply:
@scottlepp I see, thanks for explaining.